src/lang/defs/finder.rs

```diff
@@ -47,10 +47,66 @@ pub fn find_definition(
         .or_else(|| try_member_declaration(db, identifier))
         .or_else(|| try_variant_declaration(db, identifier))
         .or_else(|| try_variable_declaration(db, identifier, lookup_items))
+        .or_else(|| try_impl_items(db, identifier))
         .or_else(|| lookup_resolved_items(db, identifier, lookup_items))
         .or_else(|| lookup_item_name(db, identifier, lookup_items))
 }
 
+/// Resolve elements of `impl`s to trait definitions.
+fn try_impl_items(
+    db: &AnalysisDatabase,
+    identifier: &ast::TerminalIdentifier,
+) -> Option<ResolvedItem> {
+    if let Some(item_impl) = &identifier.as_syntax_node().ancestor_of_type::<ast::ItemImpl>(db) {
```

Comment:
I think it would be more readable since it has one lower level of nesting:
```suggestion
    let Some(item_impl) = &identifier.as_syntax_node().ancestor_of_type::<ast::ItemImpl>(db) else {
        return None;
    }
    
    // [...]  whole function code
```